_posts/2015-09-14-a-telephony-application-VoiceInn-and-its-architecture.md

```diff
@@ -0,0 +1,57 @@
+---
+layout: post
+title: "A Telephony Application - VoiceInn and Its Architecture"
+author: achyut_devkota
+modified: 2015-09-29
+tags: [telephone application, asterisk, adhearsion, sip, voip, call, IVR, PSTN, E1,]
+---
+
+VoiceInn is an internal project of YoungInnovations to provide an interactive telephony service with user friendly web interface. It includes three main telephony services - Vsurvey, Vboard and Vsupport. A modular feature of VoiceInn allows client to take service they need and also provide a customizable feature as required.
+![voiceinn logo](/images/voiceinn/voiceinn.jpg)
+
+Normally telephone service with IVR supports incoming call only but VoiceInn provides inbound as well as outbound call IVR. Outbound call can be set immediately as well as can be scheduled to a single or multiple numbers at once. 
+<!--more-->
+Any changes made by client after login  will instantly effect in telephone application when saving changes.
+
+###Services
+**Vsurvey** *- is a voice survey unlike normal survey in form.*
+
+![Vsurvey service](/images/voiceinn/vsurvey.jpg)
+
+Clients create voice survey from web interface by uploading audio file or recording from web page.  When they have done uploading audio questions they have to make a dialplan by arranging questions. There are two method of getting response from survey participants - either by audio answer or by dialing digit from keypad of phone set.
+
+Response can be collected from participants either by receiving call from participants or by calling the participants. The system can call participants individually as well as in a group. 
+
+**Vboard** *- is a Noticeboard service.*
+![Vboard service](/images/voiceinn/vboard.jpg)
+
+Clients can broadcast noticeboard by recording from phone or uploading audio file from VoiceInn website. When clients set broadcast then it becomes effective. Broadcast can be scheduled in any future time. Unlike other noticeboard services it supports outbound call as well.
```

Comment:
> When clients set broadcast then it becomes effective.

This sentence is a bit confusing.

- Broadcast can be scheduled at any time in the future.
- Supports outboud calls